### DISCUSSION TITLE
Any plans to support loading subgraphs?

### DISCUSSION THREAD

**Discussion by jamestrandung** [Cosium/spring-data-jpa-entity-graph#57] [General]
This library provides a basic feature to load direct relationship fields in an entity, which is enough to cover most of basic use cases.

````
productRepository.findById(1L, EntityGraphUtils.fromAttributePaths("brand", "maker"));
````

However, for more advanced use cases where we also need to load relationship fields of `brand` at the same time, for instance, this library doesn't support such feature yet.

This is already an existing feature of `@NamedEntityGraph` and `@NamedSubgraph`. Do we plan to support it in the future?

https://thorben-janssen.com/hibernate-tip-entitygraph-multiple-subgraphs/

**Comment by reda-alaoui** [comment]
> However, for more advanced use cases where we also need to load relationship fields of brand at the same time, for instance, this library doesn't support such feature yet.

You can already do that. Following your example, you can do:
```java
productRepository.findById(1L, EntityGraphUtils.fromAttributePaths("brand.foo", "brand.bar", "maker"));
```
where `foo` and `bar` are `Brand` attributes referencing entities.

You can also do that in a more elegant way using `spring-data-jpa-entity-graph-generator` .

> This is already an existing feature of @NamedEntityGraph and @NamedSubgraph. Do we plan to support it in the future?

If you use named EntityGraph (e.g. `EntityGraphs.named("XXX")` ), @NamedEntityGraph and @NamedSubgraph declared on the JPA Entity itself are automatically supported.

**Reply by jamestrandung** [reply]
Thanks for the information. If that `brand.foo` really work, you guys should definitely include it in the documentation. Otherwise, it's hard to know it's there. I double checked the source code and found `DynamicEntityGraph` class, which didn't look like it's capable of handling such scenarios.

`spring-data-jpa-entity-graph-generator` is another issue. I gave it a try last time but the IDE couldn't detect any builder for the graph. Perhaps, you guys can come up with a guide specially for Spring Tool Suite, which include fixes for any problems you guys encounter when using `spring-data-jpa-entity-graph-generator` for that IDE.

**Reply by reda-alaoui** [reply]
> If that brand.foo really work, you guys should definitely include it in the documentation.

I agree. Why don't you create pull request to fix that?

>  gave it a try last time but the IDE couldn't detect any builder for the graph.

I am planning to add a `spring-data-jpa-entity-graph-sample` repo showing how to install the extension with the generator on a Spring Boot application.

**Reply by reda-alaoui** [reply]
Done and done
